### DISCUSSION TITLE
FoxESS Branded Modells

### DISCUSSION THREAD

**Discussion by sleepymaxx** [nathanmarlor/foxess_modbus#431] [Ideas]
Hallo nathanmarlor, first of all im really overwhelmed by your work... i followed your and stealthCesnut´s work all the time... !
 
I myselve have two brandet inverters from FoxESS. 
They are brandet by Kopp: Kuara 10.0-3-H. The original FoxESS models are H3-10.0-E. 
So when im configuring your work (the integration itself) im getting a error: Detailed: Inverter model not supported: 'Kuara 10.0-3-H. 
So my question is if there is a chance that im getting your work to run on my home assistant also.

Every other configuration is passed and working fine. Static ip Adresses, RS232/485 to eth. ..... 
Also ive got a copy from Kopp about the modbus Protcols - witch they only copied from FoxESS direct.

PS. sorry about my bad english it was´nt used for too long and got a bit rusty. 
Thanks in advance....

**Comment by nathanmarlor** [comment]
Hey @sleepymaxx 

Thanks for the message - this repo is really just an accumulation of fantastic community knowledge and skills!

Can you share the modbus protocol document please? We'll need to confirm the same registers are available.

**Comment by sleepymaxx** [comment]
I´ve signed a document (we´re in Germany) where i confirmed to not share the modbus poll openly ----- if i´m posting this document over the internet i will be sued by the FoxESS company... so im very unsure to do this in this forum... PM doesnt function in Github... and im a very poor noob in github at all...

**Reply by nathanmarlor** [reply]
No problem, understood. Without sharing you should try to map the register addresses from the code to the document. I imagine they're very similar, if not identical to an existing model. 

If you're struggling we can find a way to share privately.

**Reply by canton7** [reply]
If I understand, op had already checked the register maps match. Surely this is just a matter of adding the inverter name to the set of supported inverters?

**Comment by sleepymaxx** [comment]
so i think Private Sharing is the best way for me... so i can kill some too openly data... and i hope you will be using the data better than me... 

@ canton yes ... i´m thinking this will do the work... but i don´t know how to..... 
BTW: it´s pretty much the same than these Adresses from this site: (And no - there are not only similar - they are complete identical) 
#https://github.com/rsaemann/HA-FoxESS-H3-Modbus/wiki/H3-Modbus-Registers

**Comment by sleepymaxx** [comment]
ps: all the Kopp branding Inverters are in origial pretty much the same than the FoxESS Inverters: 
Kuara 2.0-1-S:        S2000-G2
Kuara 3.0-1-S:        S3000-G2
Kuara 5.0-2-S:      G Serie ??
Kuara 6.0-2-T:        T6-G3
Kuara 8.0-2-T:        T8-G3
Kuara 10.0-2-T:      T10-G3
Kuara 12.0-2-T:      T12-G3
Kuara 15.0-2-T:      T15-G3
Kuara 20.0-2-T:      T20-G3
Kuara 25.0-2-T:      T25-G3
Kuara 6.0-3-H:       H3-6.0-E
Kuara 8.0-3-H:       H3-8.0-E
Kuara 10.0-3-H:     H3-10.0-E
Kuara 12.0-3-H:     H3-12.0-E

**Comment by sleepymaxx** [comment]
@ canton + @nathanmarlor are yo interested in getting them files?

**Comment by canton7** [comment]
Hi did you intend to close this discussion?

We really just need to know how the inverter identifies itself, ie what registers 30000-30015 contain. We also need to know whether the inverter has a lan port / RS485 interface, and whether it exposes input registers, holding registers, or both, over each of the available interfaces.

https://github.com/TonyM1958/HA-FoxESS-Modbus/wiki/Modbus-Map.xls is the best register map we have. Can you confirm that it matches the documents you have exactly, even down to things like fault codes?

**Comment by sleepymaxx** [comment]
Please be aware – im using a FoxESS H3 ínverter brandet in germany by kopp – These data i´ve got by kopp itself…. and this is the data i got around january 2023. So it´s not very actual…. but i think it´s no matter what…  

Thanks for your work. 

Greetings…

**Reply by sleepymaxx** [reply]
PS: sorry closed due to Lack of Understatement of the platform by me

**Reply by canton7** [reply]
Cool! Are you able to supply the information I said we needed above?

**Comment by sleepymaxx** [comment]
you got mail? Btw... i have the Kopp Kuara 10.0-3-H                    wich is similar to the FoxESS    H3-10.0-E

how can i mail you and nathanmarlor direct?

**Comment by canton7** [comment]
Can you please try the pre-release v1.8.0b1 and let me know if it works?

**Reply by sleepymaxx** [reply]
in work

**Reply by canton7** [reply]
What do you mean by "in work"?

**Reply by sleepymaxx** [reply]
never installed a pre release but will do so.. and report... but the WAF is in the moment not so high... 3 Kids Household ;-)

**Reply by canton7** [reply]
That's fine, take your time. Thanks.

**Reply by sleepymaxx** [reply]
Great

**Comment by sleepymaxx** [comment]
seems like working like a charm.... already configuring all entities and boards... you literally made my day!

**Reply by canton7** [reply]
Great, thanks for confirming! I'll get that released properly soonish

**Reply by sleepymaxx** [reply]
got the data? so i can kill it on the server?

**Reply by canton7** [reply]
I do now, thanks! Looks like it supports PV1 voltage/current/power which I didn't think it did -- I'll update at some point

**Reply by sleepymaxx** [reply]
please keep working --- your doing a great job!!! may i buy you a coffee or anything else?

**Reply by canton7** [reply]
Thanks! If you feel so inclined that would be very much appreciated. My donation links are on the right-hand side of any of my repos, e.g. https://github.com/canton7/solar_battery_forecast
